### PR TITLE
Correctly tag snapshots with z-release, add details to snapshot staging

### PR DESCRIPTION
**Pull request by schmidtd**

### CODE REVIEW

File: modules/pipeline-manifest/bin/parser.py [hunk 1]
```diff
@@ -17,9 +18,10 @@
     compenent_sha = v["git-sha256"]
     component_repo = v["git-repository"]
     component_version = v["image-tag"].replace('-'+v["git-sha256"],'')
-    if (sys.argv[2].startswith('20')):
-        # Assuming a date is coming in as a tag, mark it all up with SNAPSHOT decoration specifc to this repo
-        retag_name = component_version + "-SNAPSHOT-" + sys.argv[2]
+    if (sys.argv[2].startswith('202')):
```
Comment (ray-harris) [inline modules/pipeline-manifest/bin/parser.py:21]: is this a valid assumption? Could it start with '202' and be something else? If so, need better validation that this is a date.

Comment (schmidtd) [reply modules/pipeline-manifest/bin/parser.py:21]: It's a super-poor assumption.  We normally tag with a date (i.e. "2020-01-01-01-01-01") or with a name (i.e. "v2.0.1").  So the idea is to get just a little less poor.

Comment (ray-harris) [reply modules/pipeline-manifest/bin/parser.py:21]: I'd like to see a regex here that at least matches something like

`r"$\d{4}(-\d\d){5}^"`

We don't have to go so far as to check that it's a valid date, but be a little more careful than begins with "202".

Comment (schmidtd) [reply modules/pipeline-manifest/bin/parser.py]: Ok, all fancy'd up.  Got your begin and end strings mixed up, though. ;-)

File: modules/pipeline-manifest/bin/parser.py [hunk 2]
```diff
@@ -17,9 +18,11 @@
     compenent_sha = v["git-sha256"]
     component_repo = v["git-repository"]
     component_version = v["image-tag"].replace('-'+v["git-sha256"],'')
-    if (sys.argv[2].startswith('20')):
-        # Assuming a date is coming in as a tag, mark it all up with SNAPSHOT decoration specifc to this repo
-        retag_name = component_version + "-SNAPSHOT-" + sys.argv[2]
+    date_re=r"^\d{4}(-\d\d){5}$"
+    if re.search(date_re,sys.argv[2]):
```
Comment (ray-harris) [inline modules/pipeline-manifest/bin/parser.py:22]: I think you forgot to `import re`.

Comment (schmidtd) [reply modules/pipeline-manifest/bin/parser.py:22]: I did in my little test script, but not this one... duh

**Pull request merged**